docs/source/models.rst

```diff
@@ -1304,129 +1304,105 @@ For more info, see:
-                                                workspace=azure_workspace,
-                                                model_name='mymodelname',
-                                                service_name='myservice',
-                                                deployment_config=aci_config)
+Remarks:
+ * If ``containerResourceRequirements`` is not indicated, a deployment with minimal compute configuration is applied (``cpu: 0.1`` and ``memory: 0.5``).
+ * If ``location`` is not indicated, it defaults to the location of the workspace.
 
-    # After the model deployment completes, requests can be posted via HTTP to the new ACI
-    # webservice's scoring URI. The following example posts a sample input from the wine dataset
-    # used in the MLflow ElasticNet example:
-    # https://github.com/mlflow/mlflow/tree/master/examples/sklearn_elasticnet_wine
-    print("Scoring URI is: %s", webservice.scoring_uri)
+.. rubric:: For an AKS deployment
+
+.. code-block:: json
+
+    {
+      computeType: aks,
+      computeTargetName: aks-mlflow
+    }
+
+Remarks:
+  * In above exmaple, ``aks-mlflow`` is the name of an Azure Kubernetes Cluster registered/created in Azure Machine Learning.
+
+The following examples show how to create a deployment in ACI. Please, ensure you have `azureml-mlflow <https://pypi.org/project/azureml-mlflow/>`_ installed before continuing.
+
+.. rubric:: Example: Workflow using the Python API
+
+.. code-block:: py
 
-    import requests
     import json
-    # `sample_input` is a JSON-serialized pandas DataFrame with the `split` orientation
-    sample_input = {
-        "columns": [
-            "alcohol",
-            "chlorides",
-            "citric acid",
-            "density",
-            "fixed acidity",
-            "free sulfur dioxide",
-            "pH",
-            "residual sugar",
-            "sulphates",
-            "total sulfur dioxide",
-            "volatile acidity"
-        ],
-        "data": [
-            [8.8, 0.045, 0.36, 1.001, 7, 45, 3, 20.7, 0.45, 170, 0.27]
-        ]
+    from mlflow.deployments import get_deploy_client
+
+    # Create the deployment configuration.
+    # If no deployment configuration is provided, then the deployment happens on ACI.
+    deploy_config = {
+        "computeType": "aci"
     }
-    response = requests.post(
-                  url=webservice.scoring_uri, data=json.dumps(sample_input),
-                  headers={"Content-type": "application/json"})
-    response_json = json.loads(response.text)
-    print(response_json)
 
-.. rubric:: Example workflow using the MLflow CLI
+    # Write the deployment configuration into a file.
+    deployment_config_path = "deployment_config.json"
+    with open(deployment_config_path, "w") as outfile:
+        outfile.write(json.dumps(deploy_config))
 
-.. code-block:: bash
+    # Set the tracking uri as the deployment client.
+    client = get_deploy_client(mlflow.get_tracking_uri())
```

Comment:
Is this the recommended best practice? What is the form of the MLflow Tracking URI when used on AzureML?

In general, can we discuss the format of the AzureML deployment URI for use with `mlflow deployments`? Some of the CLI examples below refer to `<azureml-mlflow-tracking-uri>`, but they don't describe the structure of this URI and don't tell users what other AzureML deployment targets they can use (e.g. options other than the MLflow Tracking URI, if any).

Reply:
Good point, the URL can be grabbed from the Azure portal because it used to be a dynamic URL (meaning it used to be valid for 1 hour since authentication was embedded on it). Now, the URL has an specific form and interactive autentication is triggered. I think we can explain that here. On Azure ML compute, the environment variable `MLFLOW_TRACKING_URI` is automatically populated with this tracking URL and no interactive authentication will happen as user is already authenticated. So, I think I can replace `mlflow.get_tracking_uri()` by `os.environ["MLFLOW_TRACKING_URI"]` and let the user either populate that manually or if using Azure ML compute, that is done for them. Makes sense?